packages/core/src/generators/angular/signals/component.ts

```diff
@@ -0,0 +1,349 @@
+    const domRefs = getDomRefs({ json, options, rootRef, withAttributePassing });
+
+    let props: string[] = Array.from(getProps(json));
+    const events: string[] = props.filter((prop) => checkIsEvent(prop));
+    const childComponents: string[] = getChildComponents(json);
+
+    props = props.filter((prop) => {
+      // Best practise for Angular is to use Events without "on"
+      // Stencil doesn't need children as a prop
+      return prop !== 'children' && !checkIsEvent(prop);
+    });
+    const processBindingOptions: ProcessBindingOptions = {
+      events,
+      props,
+      target: 'angular',
+      skipAppendEmit: true,
+    };
+
+    options.plugins = getCodeProcessorPlugins(json, options, processBindingOptions);
+
+    if (options.plugins) {
+      json = runPostJsonPlugins({ json, plugins: options.plugins });
+    }
+
+    // CSS
+    const styles = getAngularStyles({ json, options });
+
+    // Mitosis Metadata
+    const useMetadata = json.meta?.useMetadata;
+    const onPush = useMetadata?.angular?.changeDetection == 'OnPush';
+    const writeableSignals = useMetadata?.angular?.signals?.writeable || [];
+    const requiredSignals = useMetadata?.angular?.signals?.required || [];
+
+    // Context & Injectables
+    const injectables: string[] = Object.entries(json?.context?.get || {}).map(
+      ([variableName, { name }]) => {
+        return `public ${variableName} : ${name}`;
+      },
+    );
+    const shouldUseSanitizer =
+      !useMetadata?.angular?.sanitizeInnerHTML && traverseAndCheckIfInnerHTMLIsUsed(json);
+    if (shouldUseSanitizer) {
+      injectables.push('protected sanitizer: DomSanitizer');
+    }
+
+    // HTML
+    let template = json.children
+      .map((item) => {
+        return blockToAngularSignals({
+          root: json,
+          json: item,
+          options,
+          rootRef: withAttributePassing && rootRef === ROOT_REF ? rootRef : undefined, // only pass rootRef if it's not the default
+          blockOptions: {
+            childComponents,
+            nativeAttributes: useMetadata?.angular?.nativeAttributes ?? [],
+            nativeEvents: useMetadata?.angular?.nativeEvents ?? [],
+            sanitizeInnerHTML: !shouldUseSanitizer,
+          },
+        });
+      })
+      .join('\n');
+
+    if (options.prettier !== false) {
+      template = tryFormat(template, 'html');
+    }
+
+    // Angular component settings
+    const componentsUsed = Array.from(getComponentsUsed(json)).filter(
+      (item) => item.length && isUpperCase(item[0]) && !BUILT_IN_COMPONENTS.has(item),
+    );
+    const componentSettings: Record<string, any> = {
+      selector: `'${kebabCase(json.name)}'`,
+      standalone: 'true',
+      imports: `[${['CommonModule', ...componentsUsed].join(', ')}]`,
+      template: `\`${getTemplateFormat(template)}\``,
+    };
+    if (onPush) {
+      componentSettings.changeDetection = `'ChangeDetectionStrategy.OnPush'`;
```

Comment:
http://localhost:3000/?outputTab=IYOw5grgNsBOQ%3D%3D%3D&code=FAVwzgpgshAuCGATeCAUBvYACL8B2A5iADbwBOAXFpjjgMYAW%2BBEAInBHbAJYD2eVAOQB5PAAVwDQdiwBfYLICUwYBAAeAB15lYWRBABm8EroMg8XPnixQAngGFeAWy14IeWKg1leGsIuoZMjgQMmtUGRwAHkRuADcAPkicBghiYl5kqIB6WMSZRQBuBSA%3D%3D
![Screenshot 2025-04-02 at 3 28 22 PM](https://github.com/user-attachments/assets/23416cc5-0f9f-4211-996b-e16568ae6faa)

```suggestion
      componentSettings.changeDetection = 'ChangeDetectionStrategy.OnPush';
```